### PR TITLE
TG-75: timesheet-add-edit

### PR DESCRIPTION
**Pull request by JuanBSantomero**

### CODE REVIEW

File: src/Components/TimeSheets/Add/index.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import styles from './Form.module.css';
+
+function AddTimeSheets() {
+  const [timeSheets, saveTimeSheets] = useState([]);
+  useEffect(() => {
+    fetch('http://localhost:8080/time-sheets/')
+      .then((response) => response.json())
+      .then((response) => {
+        saveTimeSheets(response.data);
+        // console.log(timeSheets);
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Add/index.js:12]: ojo el coment

File: src/Components/TimeSheets/Add/index.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import styles from './Form.module.css';
+
+function AddTimeSheets() {
+  const [timeSheets, saveTimeSheets] = useState([]);
+  useEffect(() => {
+    fetch('http://localhost:8080/time-sheets/')
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Add/index.js:8]: aca pone la variable de entorno del server

File: src/Components/TimeSheets/Add/index.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  const [employeeId, setEmployeeId] = useState('');
+  const [description, setDescription] = useState('');
+  const [project, setProject] = useState('');
+  const [date, setDate] = useState('');
+  const [hours, setHours] = useState('');
+  const [task, setTask] = useState([]);
+  const [approved, setApproved] = useState(false);
+  const [role, setRole] = useState('');
+  const addTimeSheets = async (timeSheet) => {
+    const res = await fetch('http://localhost:8080/time-sheets/', {
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Add/index.js:24]: idem, variable del server

File: src/Components/TimeSheets/Add/index.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            placeholder="Task"
+            value={task._id}
+            onChange={(e) => setTask(e.target.value)}
+          />
+        </div>
+        <div>
+          <label> Approved </label>
+          <input
+            type="checkbox"
+            // checked={approved}
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Add/index.js:126]: ojo el comment

File: src/Components/TimeSheets/Edit/index.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import styles from '../Add/Form.module.css';
+
+const params = new URLSearchParams(window.location.search);
+const timeSheetId = params.get('id');
+
+function EditTimeSheets() {
+  useEffect(() => {
+    fetch(`http://localhost:8080/time-sheets/${timeSheetId}`)
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Edit/index.js:10]: idem, variable del server

File: src/Components/TimeSheets/Edit/index.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  const [description, setDescription] = useState('');
+  const [project, setProject] = useState('');
+  const [date, setDate] = useState('');
+  const [hours, setHours] = useState('');
+  const [task, setTask] = useState([]);
+  const [approved, setApproved] = useState(false);
+  const [role, setRole] = useState('');
+
+  const editTimeSheets = async (timeSheets) => {
+    const res = await fetch(`http://localhost:8080/time-sheets/${timeSheetId}`, {
```
Comment (martinalbrnz) [inline src/Components/TimeSheets/Edit/index.js:34]: idem, variable del server

**Comment by Clarosluciano** [comment]
## Testing Results

Tested locally on: https://alfonso-trackgenix-app-git-tg-76-list-timeshe-5a11d0-basp-m2022.vercel.app

Flagged checklists in Trello Ticket aren't reproducible from the tested environment. 

Test Results: Failed 

## Bugs found:

Scenario 1:
1. **Given** the user clicks on "Timesheets" button
image

2. **When** clicks on "Add" button
image

3. **Then** the user is redirect to https://alfonso-trackgenix-server.vercel.app/time-sheets/. The API is returning all timesheets
image

Scenario 2:
1. **Given** the user clicks on "Timesheets" button
image

2. **When** clicks on any "Edit" button
image

3. **Then** the user is redirect to https://alfonso-trackgenix-server.vercel.app/time-sheets/null. The API is returning a error
image

Note: I cloned both repositories and merged the content of both branches to do this test.

**Pull request closed without merging**